### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by jonasteuwen**
This branch improves the test coverage of dlup

### CODE REVIEW

File: tests/backends/test_tifffile_backend.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        assert slide.vendor is None
+        assert slide.magnification is None
+
+        # Check the properties
+
+    def test_read_region_invalid_level(self, mock_tifffile_slide):
+        slide = mock_tifffile_slide
+        with pytest.raises(RuntimeError, match="Level 4 not present."):
+            slide.read_region((0, 0), 4, (100, 100))
```
Comment (EricMarcus-ai) [inline tests/backends/test_tifffile_backend.py:57]: Should we also test the read_region itself? I.e. whether it returns correct dimensions etc

File: tests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    with pytest.raises(ArgumentTypeError):
+        file_path("/path/which/does/not/exist.txt")
+
+
+def test_file_path_no_need_exists():
+    _path = "/path/which/does/not/need/to/exist.txt"
+    assert file_path(_path, need_exists=False) == pathlib.Path(_path)
+
+
+def test_main_no_arguments(capsys):
```
Comment (EricMarcus-ai) [inline tests/test_cli.py:36]: Is capsys needed for catching a systemexit?

**Review by EricMarcus-ai — COMMENTED** [review]
Nice improvements to the testing. Some minor comments.

File: dlup/backends/common.py [hunk 3]
```diff
@@ -160,6 +159,17 @@ def get_thumbnail(self, size: int | tuple[int, int]) -> PIL.Image.Image:
 
         downsample = max(*(dim / thumb for dim, thumb in zip(self.dimensions, size)))
         level = self.get_best_level_for_downsample(downsample)
+
+        # Debugging logs:
```
[the remark this answers is not in the document]
Comment (jonasteuwen) [reply dlup/backends/common.py]: Don't know how that ended up there. Fixed.

File: dlup/experimental_backends/pyvips_backend.py [hunk 4]
```diff
@@ -191,7 +190,8 @@ def associated_images(self):
         """Images associated with this whole-slide image."""
         if not self._loader == "openslideload":
             return {}
-        associated_images = (_.strip() for _ in self.properties["slide-associated-images"].split(","))
+        # TODO: Fix this
```
[the remark this answers is not in the document]
Comment (jonasteuwen) [reply dlup/experimental_backends/pyvips_backend.py:193]: It doesn't properly load them yet, it's also in 'experimental_backends' so I'm going to keep it like this. There is #151 and other PRs that need to fix this.

File: tests/backends/test_common.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+import numpy as np
+import pytest
+from PIL import Image
+
+from dlup.backends.common import AbstractSlideBackend, numpy_to_pil
+
+
+def test_numpy_to_pil_single_channel():
+    arr = np.random.randint(0, 256, (10, 10, 1), dtype=np.uint8)
```
[the remark this answers is not in the document]
Comment (jonasteuwen) [reply tests/backends/test_common.py]: Alright, done

File: tests/backends/test_common.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    def test_spacing_without_set(self):
+        slide = self.DummySlideBackend("test_filename.tiff")
+        slide._spacings = None
+        assert slide.spacing is None
+
+    def test_get_thumbnail(self):
+        slide = self.DummySlideBackend("test_filename.tiff")
+
+        # Getting a 200x200 thumbnail
+        thumbnail = slide.get_thumbnail(200)
```
[the remark this answers is not in the document]
Comment (jonasteuwen) [reply tests/backends/test_common.py:107]: There is a type defined right? So mypy should complain

File: tests/backends/test_tifffile_backend.py (hunk 1 above)
Comment (jonasteuwen) [reply tests/backends/test_tifffile_backend.py:57]: That depends on the backend, which is mocked, so I'd say no.

**Review by EricMarcus-ai — APPROVED** [review]
LGTM

**Pull request merged**